Support/Interfaces/Core/Repository.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file Support/Interfaces/Core/Repository.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ * @ingroup support
+ *
+ * @brief Interface for Repository Classes
+ */
+
+namespace PKP\Support\Interfaces\Core;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+
+interface Repository
```

Comment:
As I understand it, Illuminate doesn't normally have a Repository pattern, but it can be useful to borrow some of those conventions from e.g. Symfony (https://www.twilio.com/blog/repository-pattern-in-laravel-application). Since this class doesn't extend a base class, and since the particular conventions we use are arbitrary, I wonder whether we shouldn't make the functions here align a little more closely with our own `Repository` pattern. For example, `get` instead of `find`, `add` instead of `create`, `edit` instead of `update`, etc. Otherwise it's a whole set of synonyms to remember.